tests/www/views/test_views_dataset.py

```diff
@@ -0,0 +1,281 @@
+            ("last_dataset_update", [3, 2, 4, 1]),
+            ("-last_dataset_update", [1, 4, 2, 3]),
+        ],
+    )
+    def test_order_by(self, admin_client, session, order_by, ordered_dataset_ids):
+        datasets = [
+            DatasetModel(
+                id=i,
+                uri=string.ascii_lowercase[i],
+                extra={"foo": "bar"},
+                created_at=timezone.parse(self.default_time),
+                updated_at=timezone.parse(self.default_time),
+            )
+            for i in range(1, len(ordered_dataset_ids) + 1)
+        ]
+        session.add_all(datasets)
+        dataset_events = [
+            DatasetEvent(
+                dataset_id=datasets[2].id,
+                timestamp=pendulum.today('UTC').add(days=-3),
+            ),
+            DatasetEvent(
+                dataset_id=datasets[1].id,
+                timestamp=pendulum.today('UTC').add(days=-2),
+            ),
+            DatasetEvent(
+                dataset_id=datasets[1].id,
+                timestamp=pendulum.today('UTC').add(days=-1),
+            ),
+        ]
+        session.add_all(dataset_events)
+        session.commit()
+        assert session.query(DatasetModel).count() == len(ordered_dataset_ids)
+
+        response = admin_client.get(f"/object/list_datasets?order_by={order_by}")
+
+        assert response.status_code == 200
+        assert ordered_dataset_ids == [json_dict['id'] for json_dict in response.json['datasets']]
+        assert response.json['total_entries'] == len(ordered_dataset_ids)
+
+    @pytest.mark.need_serialized_dag
+    def test_correct_counts_update(self, admin_client, session, dag_maker, app, monkeypatch):
+        with monkeypatch.context() as m:
+            datasets = [Dataset(uri=f"s3://bucket/key/{i}") for i in [1, 2]]
+
+            with dag_maker(dag_id='downstream', schedule=datasets, serialized=True, session=session):
+                EmptyOperator(task_id='task1')
+
+            with dag_maker(dag_id='upstream', schedule=None, serialized=True, session=session):
+                EmptyOperator(task_id='task1', outlets=[datasets[0]])
+
+            m.setattr(app, 'dag_bag', dag_maker.dagbag)
+
+            ds1_id = session.query(DatasetModel.id).filter_by(uri=datasets[0].uri).scalar()
+            ds2_id = session.query(DatasetModel.id).filter_by(uri=datasets[1].uri).scalar()
+
+            session.add(
+                DatasetEvent(
+                    dataset_id=ds1_id,
+                    timestamp=pendulum.DateTime(2022, 8, 1, tzinfo=UTC),
+                )
+            )
+            session.add(
+                DatasetEvent(
+                    dataset_id=ds1_id,
+                    timestamp=pendulum.DateTime(2022, 8, 1, tzinfo=UTC),
+                )
+            )
+            session.add(
+                DatasetEvent(
+                    dataset_id=ds1_id,
+                    timestamp=pendulum.DateTime(2022, 8, 1, tzinfo=UTC),
+                )
+            )
```

Comment:
```suggestion
            session.add_all([
                DatasetEvent(
                    dataset_id=ds1_id,
                    timestamp=pendulum.DateTime(2022, 8, 1, tzinfo=UTC),
                ) for i in range(3)
            ])
```

nit: similar to above.